caching-big-repositories_5bf03fb92c7d3a31944e139d.md

```diff
@@ -0,0 +1,167 @@
+
+* [Overview](#overview)
+* [The scenario](#the-scenario)
+* [The logic behind the recipe](#the-logic-behind-the-recipe)
+* [An example project](#an-example-project)
+* [See Also](#see-also)
+
+## Overview
+
+The purpose of this document is to provide a recipe for caching big
+repositories in order to save time.
+
+## The scenario
+
+Imagine that you have one or multiple large files on a GitHub repository and
+that you want to make the downloading of them as fast as possible.
+
+## The logic behind the recipe
+
+Using the Cache server provided by Semaphore 2.0 for getting the files of your
+GitHub repository is much faster than using the GitHub server machines for the
+same task because the Semaphore Cache server is on the same network as the
+Virtual Machines used for executing the jobs of a Semaphore pipeline.
+
+What the example project that follows will do is getting all the files of the
+GitHub repository in the first job using `checkout` and then storing them into
+the Semaphore 2.0 Cache server using `cache store`. The remaining jobs of the
+project will use these files from the Cache server using `cache restore`
+without calling `checkout` again.
+
+Note that for this recipe to work, the files of the GitHub repository should
+remain intact through the lifetime of the Semaphore project.
+
+## An example project
+
+The example project of this section will use 10 large files. Each one of these
```

Comment:
We don't need to provide any details about example project. Proceed straight to a configuration that people can copy and paste.